### PR TITLE
More host image copy validation

### PR DESCRIPTION
**Pull request by Tony-LunarG**
Addresses most of #6317

### CODE REVIEW

File: layers/core_checks/cc_copy_blit_resolve.cpp [hunk 1]
```diff
@@ -2817,6 +2817,8 @@ bool CoreChecks::ValidateMemoryImageCopyCommon(VkDevice device, InfoPointer info
     const char *image_layout_vuid = from_image ? "VUID-VkCopyImageToMemoryInfoEXT-srcImageLayout-09064"
                                                : "VUID-VkCopyMemoryToImageInfoEXT-dstImageLayout-09059";
 
+    // VUID-VkMemoryToImageCopyEXT-pHostPointer-09061 and VUID-VkImageToMemoryCopyEXT-pHostPointer-09066 are tested as best as
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_copy_blit_resolve.cpp:2820]: the validation stats script doesn't count things in comments

we have a `unimplementable_validation.h` file you can put these 2 there with details

File: layers/core_checks/cc_copy_blit_resolve.cpp [hunk 2]
```diff
@@ -2903,6 +2912,41 @@ bool CoreChecks::ValidateMemoryImageCopyCommon(VkDevice device, InfoPointer info
...
+                }
+                const void *copy_end = static_cast<const char *>(region.pHostPointer) + copy_size;
+
+                if ((region.pHostPointer >= state->p_driver_data && region.pHostPointer < mapped_end) ||
+                    (copy_end >= state->p_driver_data && copy_end < mapped_end) ||
+                    (region.pHostPointer <= state->p_driver_data && copy_end > mapped_end)) {
+                    const char *vuid =
+                        from_image ? "VUID-VkImageToMemoryCopyEXT-pRegions-09067" : "VUID-VkMemoryToImageCopyEXT-pRegions-09062";
+                    LogObjectList objlist(device, image_state->image());
+                    skip |= LogError(vuid, objlist, loc,
```
Comment (spencer-lunarg) [inline layers/core_checks/cc_copy_blit_resolve.cpp:2940]: Use `Location` to point to the variable

```
skip |= LogError(vuid, objlist,  loc.dot(Field::pRegions, i).dot(Field::pHostPointer), "points to memory ...."
```

**Review by spencer-lunarg — APPROVED** [review]
LGTM - Just address the 2 nit comments please

File: layers/core_checks/cc_copy_blit_resolve.cpp (hunk 1 above)
Comment (Tony-LunarG) [reply layers/core_checks/cc_copy_blit_resolve.cpp]: Moved them there

File: layers/core_checks/cc_copy_blit_resolve.cpp (hunk 2 above)
Comment (Tony-LunarG) [reply layers/core_checks/cc_copy_blit_resolve.cpp]: That's pretty cool!  Done

**Pull request merged**